### PR TITLE
Android: Fix lineCap of Polyline

### PR DESCRIPTION
**Pull request by kruyvanna**
### Does any other open PR do the same thing?

No

### What issue is this PR fixing?

https://github.com/react-community/react-native-maps/issues/2374

### How did you test this PR?

Verified on Android, HTC M8

### CODE REVIEW

**Comment by rborn** [comment]
Hi, thanks for the PR ❤️
 
Does it work correctly on ios? (both ios and google maps)

**Comment by kruyvanna** [comment]
Hi @rborn, on iOS, the MapKit version works correctly.
for GoogleMaps on iOS, the native SDK doesn't support lineCap unfortunately (as of today).
https://developers.google.com/maps/documentation/ios-sdk/shapes

**Comment by kruyvanna** [comment]
@rborn I send you the screenshots of iOS and Android
line_cap

**Comment by rborn** [comment]
@kruyvanna that's great, one more thing: could you please amend the  docs that it's not supported yet? https://github.com/react-community/react-native-maps/blob/master/docs/polyline.md

**Comment by kruyvanna** [comment]
@rborn done. Not sure about my English wording.

File: lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolyline.java [hunk 1]
```diff
@@ -22,6 +24,7 @@
   private float width;
   private boolean geodesic;
   private float zIndex;
+  private Cap lineCap = new RoundCap();
```
Comment (rborn) [inline lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolyline.java]: Why do we need here a RoundCap ?

File: docs/polyline.md [hunk 2]
```diff
@@ -9,7 +9,7 @@
 | `strokeColor` | `String` | `#000` | The stroke color to use for the path.
 | `strokeColors` | `Array<String>` | `null` | The stroke colors to use for the path (iOS only). Must be the same length as `coordinates`.
 | `lineCap` | `String` | `round` | The line cap style to apply to the open ends of the path. Possible values are `butt`, `round` or `square`.
-| `lineJoin` | `String` | `round` | The line join style to apply to corners of the path. Possible values are `miter`, `round` or `bevel`.
+| `lineJoin` | `String` | `round` | The line join style to apply to corners of the path. Possible values are `miter`, `round` or `bevel`. Note: lineJoin is not yet supported for GoogleMaps provider on iOS.
```
Comment (rborn) [inline docs/polyline.md]: It's `lineJoin` not available or `lineCap` (english is great 🤗)

Comment (kruyvanna) [reply docs/polyline.md]: oops!, it's lineCap. :D

File: lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolylineManager.java [hunk 3]
```diff
@@ -68,6 +72,26 @@ public void setZIndex(AirMapPolyline view, float zIndex) {
...
+        cap = new ButtCap();
+        break;
+      case "round":
+        cap = new RoundCap();
+        break;
+      case "square":
+        cap = new SquareCap();
+        break;
+      default:
+        cap = new RoundCap();
```
Comment (rborn) [inline lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolylineManager.java]: Dow e need a default cap ? I mean maybe people don't want one.

Comment (kruyvanna) [reply lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolylineManager.java]: the docs say round is the default one

**Comment by kruyvanna** [comment]
@rborn I corrected the docs changing from lineJoin to lineCap

File: lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolylineManager.java (hunk 3 above)
Comment (rborn) [reply lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolylineManager.java]: ok great, I'm blind 😹

**Comment by rborn** [comment]
All good, thanks a lot ❤️

@alvelig LGTM 🐽

**Comment by kruyvanna** [comment]
@rborn Btw, I added support for lineDashPattern on Android. 
The appearance of the dash is dependent on lineCap property.
So, should I wait for this PR to be merged first then make a new PR or should I make a new PR right a way.
https://github.com/kruyvanna/react-native-maps/commit/f0ac2e5996e6b1f2e7f1ac1c1a11e2a669ab45f9

File: lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolylineManager.java (hunk 3 above)
Comment (kruyvanna) [reply lib/android/src/main/java/com/airbnb/android/react/maps/AirMapPolylineManager.java]: :D

**Comment by rborn** [comment]
@kruyvanna new PR please

**Comment by kruyvanna** [comment]
@rborn ok

**Pull request merged**